pkg/status/toolchaincluster.go

```diff
@@ -19,18 +19,20 @@ const (
 
 // ToolchainClusterAttributes required attributes for obtaining ToolchainCluster status
 type ToolchainClusterAttributes struct {
-	GetClusterFunc func() (*cluster.CachedToolchainCluster, bool)
-	Period         time.Duration
-	Timeout        time.Duration
+	GetClustersFunc cluster.GetClustersFunc
```

Comment:
the purpose and the usage of the `ToolchainClusterAttributes` type and `GetToolchainClusterConditions` function doesn't match with the `cluster.GetClustersFunc` - this is intended to be used for one ToolchainCluster, but for many. I saw that you used 
```go
toolchainCluster := toolchainClusters[0]
```
to get the first one, but you never know if you are supposed to use the first one, second, or third.
I would rather either keep the 
```go
GetClusterFunc func() (*cluster.CachedToolchainCluster, bool)
```
in the same format is it was before and let the caller to pass the ToolchainCluster - something like this:
```go
	attributes := status.ToolchainClusterAttributes{
		GetClusterFunc: func() (*cluster.CachedToolchainCluster, bool) {
			clusters := cluster.GetClustersFunc()
			return clusters[0], len(clusters) != 0
		},
```
or just change the field so it doesn't take a func but rather a `*cluster.CachedToolchainCluster` directly